### PR TITLE
refactor code

### PR DESCRIPTION
**Pull request by robatipoor**

### CODE REVIEW

**Comment by dgriffen** [comment]
No idea why the CI build failed, it worked when I ran it manually. Out of curiosity, since most of these are formatting changes, were these applied with rustcfmt? and if so which version?

**Comment by robatipoor** [comment]
I do not know why azure-pipelines Could not find /azure-pipelines/build-template.yml 
yes i use rustfmt version 1.2.2-nightly (5274b49 2019-04-24)

File: dotenv/src/errors.rs [hunk 1]
```diff
@@ -1,16 +1,37 @@
...
+        )
+    }
+}
+
+impl error::Error for ParseError {}
 
 #[derive(Debug)]
 pub enum Error {
-    LineParse(String, usize),
+    LineParse(ParseError),
```
Comment (dgriffen) [inline dotenv/src/errors.rs]: What is the logic behind introducing a new layer of indirection here?

Comment (robatipoor) [reply dotenv/src/errors.rs]: i think this enhance readability and Integrity code what you think ?

**Comment by ZoeyR** [comment]
Closing since the code has undergone a lot of changes since this was opened.

**Pull request closed without merging**